client/src/app/site/deployment-center/deployment-center-setup/wizard-logic/deployment-center-state-manager.ts

```diff
@@ -227,6 +239,39 @@ export class DeploymentCenterStateManager implements OnDestroy {
       return Observable.of(r.id);
     });
   }
+
+  private _canCreateAadApp(): Observable<{ status: string; statusMessage: string; result: any }> {
+    const success = {
+      status: 'succeeded',
+      statusMessage: null,
+      result: null,
+    };
+
+    const permissionPayload: PermissionsResultCreationParameters = {
+      aadPermissions: {
+        token: this._vstsApiToken,
+        tenantId: parseToken(this._token).tid,
+      },
+    };
+
+    if (!this._canCreateApp) {
+      return this._azureDevOpsService.getPermissionResult(permissionPayload).map((r: PermissionsResult) => {
```

Comment:
Should we cache this for a session?

Reply:
I am caching it in the state-manager via this._canCreateApp. But session cache makes sense. Let me see if I can do that.

Reply:
Added caching